skimage/transform/_geometric.py

```diff
@@ -987,76 +1058,56 @@ class AffineTransform(ProjectiveTransform):
     def __init__(
         self,
         matrix=None,
+        *,
```

Comment:
Same here. Turning arguments following `matrix=` keyword-only makes sense, but breaks users without warning. It's also inconsistent with `EssentialMatrixTransform` which makes everything keyword-only?

We should probably check that these things are consistent.

Reply:
I would argue that `EssentialMatrixTransform` is consistent, in that the rule is "all transformation parameters are keyword only except an input homogeneous matrix".    `EssentialMatrixTransform` does not accept a matrix input.